src/lazy/encoder/text/v1_0/value_writer.rs

```diff
@@ -28,67 +28,13 @@ pub struct TextValueWriter_1_0<'value, W: Write + 'value> {
     pub(crate) parent_type: ParentType,
 }
 
-/// Returns `true` if the provided `token`'s text is an 'identifier'. That is, the text starts
-/// with a `$`, `_` or ASCII letter and is followed by a sequence of `$`, `_`, or ASCII letters
-/// and numbers. Examples:
-/// * `firstName`
-/// * `first_name`
-/// * `name_1`
-/// * `$name`
-/// Unlike other symbols, identifiers don't have to be wrapped in quotes.
-fn token_is_identifier(token: &str) -> bool {
-    if token.is_empty() {
-        return false;
-    }
-    let mut chars = token.chars();
-    let first = chars.next().unwrap();
-    (first == '$' || first == '_' || first.is_ascii_alphabetic())
-        && chars.all(|c| c == '$' || c == '_' || c.is_ascii_alphanumeric())
-}
```

Comment:
🗺️ The methods in this section were originally copied from the `IonValueFormatter`. This PR introduces the `IoFmtShim` type that allows them to be re-used instead of copied.